pkg/ottl/ottlfuncs/func_parse_key_value_test.go

```diff
@@ -0,0 +1,374 @@
+			assert.True(t, ok)
+
+			expected := pcommon.NewMap()
+			assert.NoError(t, expected.FromRaw(tt.expected))
+
+			assert.Equal(t, expected.Len(), actual.Len())
+			expected.Range(func(k string, v pcommon.Value) bool {
+				ev, _ := expected.Get(k)
+				av, ok := actual.Get(k)
+				assert.True(t, ok)
+				assert.Equal(t, ev, av)
+				return true
+			})
+		})
+	}
+}
+
+func Test_parseKeyValue_equal_delimiters(t *testing.T) {
+	target := ottl.StandardStringGetter[any]{
+		Getter: func(ctx context.Context, tCtx any) (any, error) {
+			return "", nil
+		},
+	}
+	delimiter := ottl.NewTestingOptional[string]("=")
+	pairDelimiter := ottl.NewTestingOptional[string]("=")
+	_, err := parseKeyValue[any](target, delimiter, pairDelimiter)
+	assert.Error(t, err)
+
+	delimiter = ottl.NewTestingOptional[string](" ")
+	_, err = parseKeyValue[any](target, delimiter, ottl.Optional[string]{})
+	assert.Error(t, err)
+}
+
+func Test_parseKeyValue_bad_target(t *testing.T) {
+	target := ottl.StandardStringGetter[any]{
+		Getter: func(ctx context.Context, tCtx any) (any, error) {
+			return 1, nil
+		},
+	}
+	delimiter := ottl.NewTestingOptional[string]("=")
+	pairDelimiter := ottl.NewTestingOptional[string]("!")
+	exprFunc, err := parseKeyValue[any](target, delimiter, pairDelimiter)
+	assert.NoError(t, err)
+	_, err = exprFunc(context.Background(), nil)
+	assert.Error(t, err)
+}
+
+func Test_parseKeyValue_empty_target(t *testing.T) {
+	target := ottl.StandardStringGetter[any]{
+		Getter: func(ctx context.Context, tCtx any) (any, error) {
+			return "", nil
+		},
+	}
+	delimiter := ottl.NewTestingOptional[string]("=")
+	pairDelimiter := ottl.NewTestingOptional[string]("!")
+	exprFunc, err := parseKeyValue[any](target, delimiter, pairDelimiter)
+	assert.NoError(t, err)
+	_, err = exprFunc(context.Background(), nil)
+	assert.Error(t, err)
+}
+
+func Test_parseKeyValue_bad_split(t *testing.T) {
+	target := ottl.StandardStringGetter[any]{
+		Getter: func(ctx context.Context, tCtx any) (any, error) {
+			return "name=ottl!hello_world", nil
+		},
+	}
+	delimiter := ottl.NewTestingOptional[string]("=")
+	pairDelimiter := ottl.NewTestingOptional[string]("!")
+	exprFunc, err := parseKeyValue[any](target, delimiter, pairDelimiter)
+	assert.NoError(t, err)
+	_, err = exprFunc(context.Background(), nil)
+	assert.ErrorContains(t, err, "cannot split \"hello_world\" into 2 items, got 1 item(s)")
+}
+
+func Test_parseKeyValue_mismatch_quotes(t *testing.T) {
+	target := ottl.StandardStringGetter[any]{
+		Getter: func(ctx context.Context, tCtx any) (any, error) {
+			return `k1=v1 k2='v2"`, nil
+		},
+	}
+	exprFunc, err := parseKeyValue[any](target, ottl.Optional[string]{}, ottl.Optional[string]{})
+	assert.NoError(t, err)
+	_, err = exprFunc(context.Background(), nil)
+	assert.Error(t, err)
+}
+
+func Test_parseKeyValue_bad_delimiter(t *testing.T) {
+	target := ottl.StandardStringGetter[any]{
+		Getter: func(ctx context.Context, tCtx any) (any, error) {
+			return "a=b c=d", nil
```

Comment:
Could we have a similar tets for `a==b c=d` or something similar that would produce too many pairs?

Reply:
Got it, that makes sense. I think we are good here then, thank you.